sources/lib/util/options.ml

```diff
@@ -381,7 +420,24 @@ let mk_execution_opt frontend input_format parse_only parsers
     preludes no_locs_in_answers type_only type_smt2
   =
   let answers_with_loc = not no_locs_in_answers in
-  let input_format = "." ^ input_format in
+
+  vinfer_input_format :=
+    (match input_format with Some _ -> false | None -> true);
+  let input_format : input_format = match input_format with
+    | None -> Native
+    | Some fmt ->
+      match match_input_format fmt with
+      | Native -> Native
+      | Smtlib2 -> Smtlib2
+      | Why3 ->
+        Format.eprintf "[warning] Why3 format is only available when \
+                        AB-Why3 parser is loaded@.";
+        Why3
```

Comment:
If I understand correctly, this warning will *always* appear even when the why3 parser has indeed been loaded, right ?
I think this is wrong and that a better behaviour would be the following: when the why3 parser is loaded no warning should appear, and when it isn't loaded, this should be an error.

Reply:
That was in the first version, but since you didn't want me to use Str to search in the dynamically linked parser name "why3" i removed it.

Reply:
Well, altogether this strongly suggests that here (during parsing of options), is a really bad place to emit a warning like this...
This warning should rather be emitted in the parsers file, where we can know whether we have a specific parser for why3 (coming from the AB-why3 plugin or another provided by the user), or if we don't have one, and we instead try and fallback on the native format parser, while emitting a warning that it might be an error (if the provided file is in why3 and not alt-ergo native format), and that anyway the extension has been deprecated for alt-ergo native files.

Reply:
Since .why and .mlw are set to depreciated, i've remove them from known extension of Alt-Ergo native parser. If this extension are found the match_extension will return the Why3 format. At parsing, if no parser exist for why3 we will print a warning telling user that they should use the AB-Why3 plugin and this extension are depreciated, and then continue with native parser. Does this sound good for you ?

Reply:
Sounds good indeed.